src/zc/buildout/easy_install.py

```diff
@@ -733,7 +736,20 @@ def install(self, specs, working_set=None):
                     pkg_resources.VersionConflict(dist, req), ws)
 
             best[req.key] = dist
-            extra_requirements = dist.requires(req.extras)[::-1]
+            if self._allow_unknown_extras:
```

Comment:
I'd add a comment on the line below like this:

    # Note: by default, self._allow_unknown_extras is False

Reason: the "if" part is normally the normal path, but in this case it is the "else" part that's almost always executed.

Reply:
Another idea, which just means shuffling some code around: there is now a good warning, but the error is the same.

You could use the `missing_requested` list in both cases.

    if missing_requested:
        if self._allow_unkown_extras:
            logger.warn(...)
        else:
            logger.error(..., you can use --allow-unkown-extras if that is not a problem)
            sys.exit(1)   # or however that happens in the rest of the code